src/components/Identity/createComponent.js

```diff
@@ -150,6 +129,27 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
     commands: {
       setCustomerIds(options) {
         return customerIds.sync(options);
+      },
+      getEcid() {
+        if (ecid) {
+          return ecid;
+        }
+        if (deferredForIdentityCookie) {
+          // Should execute only after the first call resloves
+          return accommodateIdentityOnRequest().then(() => ecid);
+        }
+        if (!hasIdentityCookie()) {
+          // Should defer all calls if an
+          deferredForIdentityCookie = defer();
+        }
```

Comment:
My biggest concern is that understanding how the deferred and `accommodateIdentityOnRequest` and `getEcidCommand` all coordinate together is getting somewhat complex. I would need to dig into it for a while to see how we can simplify things, but it's on my mind. I don't think it needs to hold up the PR.